radix-engine/src/model/nodes/transaction_processor.rs

```diff
@@ -250,6 +250,12 @@ impl TransactionProcessor {
         let auth_zone_id: AuthZoneId = auth_zone_ref.into();
 
         for inst in input.instructions.as_ref() {
+            system_api
+                .emit_application_event(ApplicationEvent::PreExecuteInstruction {
+                    instruction: &inst,
```

Comment:
Good idea :+1:

Reply:
(Also a note to self: there's a corner case when handling any sys calls that happen after all instructions have been processed. Their associated index should be None, instead of the latest idx)

Reply:
_(addressed the corner case too)_